### PR TITLE
feat: add support for grug-far.nvim

### PR DESCRIPTION
**Pull request by sjclayton**
Add support for [grug-far.nvim](https://github.com/MagicDuck/grug-far.nvim) - a new super simple find and replace plugin powered by ripgrep.

### CODE REVIEW

File: lua/rose-pine.lua [hunk 1]
```diff
@@ -126,7 +126,7 @@ local function set_highlights()
 		RedrawDebugClear = { fg = palette.base, bg = palette.gold },
 		RedrawDebugComposed = { fg = palette.base, bg = palette.pine },
 		RedrawDebugRecompose = { fg = palette.base, bg = palette.love },
-		Search = { fg = palette.base, bg = palette.text },
+		Search = { fg = palette.base, bg = palette.rose },
```
Comment (mvllow) [inline lua/rose-pine.lua:129]: Let's leave this how it was, having two colours for search and current search is a little overwhelming in my opinion

File: lua/rose-pine.lua [hunk 2]
```diff
@@ -941,6 +941,22 @@ local function set_highlights()
...
+		GrugFarHelpWinActionKey = { fg = palette.gold },
+		GrugFarHelpWinActionPrefix = { fg = palette.foam },
+		GrugFarHelpWinActionText = { fg = palette.pine },
+		GrugFarHelpWinHeader = { link = "FloatTitle" },
+		GrugFarInputLabel = { fg = palette.foam },
+		GrugFarInputPlaceholder = { link = "Comment" },
+		GrugFarResultsActionMessage = { fg = palette.foam },
+		GrugFarResultsChangeIndicator = { fg = groups.git_change },
+		GrugFarResultsHeader = { fg = palette.pine },
+		GrugFarResultsMatch = { link = "Search" },
```
Comment (mvllow) [inline lua/rose-pine.lua:957]: If we want the grug matches to be more prominent we can link to the "CurSearch" instead

```suggestion
		GrugFarResultsMatch = { link = "CurSearch" },
```

**Review by mvllow — CHANGES_REQUESTED** [review]

File: lua/rose-pine.lua (hunk 2 above)
Comment (sjclayton) [reply lua/rose-pine.lua]: That works.

File: lua/rose-pine.lua (hunk 1 above)
Comment (sjclayton) [reply lua/rose-pine.lua:129]: May I suggest a modification to this for the theme overall, as I feel the default background for `Search` is terrible. The white `text` seems to glaring to me, the gold for `CurSearch` is ok, but could we try toning down the overall results background?

Comment (mvllow) [reply lua/rose-pine.lua:129]: Yea that's completely fine. Let's discuss in a new issue https://github.com/rose-pine/neovim/issues/305

**Review by mvllow — APPROVED** [review]

**Pull request merged**